src/core/app/services/project-report.services.ts

```diff
@@ -0,0 +1,103 @@
+import { Report, Project, Task, ProjectStatistics } from '../interfaces/project-report.interface';
+import { ProjectRepository } from '../../infra/repositories/project.repository';
+import { EmployeeTaskRepository } from '../../infra/repositories/employee-task.repository';
+import { EmployeeRepository } from '../../infra/repositories/employee.repository';
+import { TaskRepository } from '../../infra/repositories/tasks.repository';
+import { CompanyRepository } from '../../infra/repositories/company.repository';
+
+/**
+ * @brief Esta función inicializa las estadísticas de un reporte
+ * 
+ * @param total: Number 
+ * 
+ * @return ProjectStatistics
+ * 
+ * @description Las tareas se agrupan segun su estatus y se inicializan en 0.
+ * 
+ */
+function initilizeStatistics(total: Number): ProjectStatistics { 
+    const raw : ProjectStatistics = { 
+        total: total,
+        done: 0,
+        inprogress: 0,
+        underrevision: 0,
+        delayed: 0,
+        postponed: 0,
+        notstarted: 0,
+        cancelled: 0,
+    }
+    return raw;
+}
+
+
+/**
+ * @brief Esta función genera el reporte de un proyecto
+ * 
+ * @param id: string 
+ * 
+ * @return Report
+ * 
+ * @description Se hace la distinción entre un proyecto en cotización o 
+ * cancelado y un proyecto en cualquier otro estatus. El reporte de un proyecto en cotización
+ * o cancelado retorna datos generales del proyecto y los demás incluyen tareas asociadas
+ * al proyecto y sus estadísticas.
+ * 
+ */
+async function getReport(id: string): Promise<Report> {
+    try {
+        let report : Report;
+
+        const status = await ProjectRepository.findProjectStatusById(id);
+        const rawProject = await ProjectRepository.findById(id);
+        const company = await CompanyRepository.findById(rawProject.idCompany);
+
+        const project: Project = {...rawProject, companyName: company.name};
+        report = {project: project};
+
+       
+        if (status.status !== 'In quotation' && status.status !== '-' && status.status !== 'Cancelled'){  // 'Accepted' 'Not started' 'In process' 'Under revision' 'Delayed' 'Postponed' 'Done'
```

Comment:
En vez de hardcodear los strings mejor usen constantes. @DHurtado714-itesm ¿Desde dónde podrían crear las constantes `In quotation`, `Cancelled`, etc... para que se estandarice?